base_search_mail_content/models/mail_thread.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+# © 2016-17 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# © 2016 Serpent Consulting Services Pvt. Ltd. (<http://www.serpentcs.com>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from lxml import etree
+from odoo.osv import expression
+from odoo.osv.orm import setup_modifiers
+
+
+class MailThread(models.AbstractModel):
+
+    _inherit = 'mail.thread'
+
+    def _search_message_content(self, operator, value):
+
+        main_operator = 'in'
+        if operator in expression.NEGATIVE_TERM_OPERATORS:
+            main_operator = 'not in'
+            operators = {'!=': '=', 'not like': 'like',
+                         'not ilike': 'ilike', 'not in': 'in'}
+            operator = operators[operator]
+        domain = [('model', '=', self._name), '|', '|', '|', '|',
```

Comment:
It seems to me that it would be much cleaner and more efficient to do something like:

```python
domain = [('model', '=', self._name)]
if operator not in expression.NEGATIVE_TERM_OPERATORS:
    domain += ["|"] * 4
domain += [('record_name', operator, value),
           ('subject', operator, value), 
           ('body', operator, value),
           ('email_from', operator, value),
           ('reply_to', operator, value)]
```